akka-actor-typed/src/main/scala/akka/actor/typed/internal/StashBufferImpl.scala

```diff
@@ -167,8 +97,8 @@ import akka.util.ConstantFun
   override def unstash(scaladslCtx: scaladsl.ActorContext[T], behavior: Behavior[T],
                        numberOfMessages: Int, wrap: T ⇒ T): Behavior[T] = {
     val iter = new Iterator[T] {
-      override def hasNext: Boolean = MutableStashBufferImpl.this.nonEmpty
-      override def next(): T = MutableStashBufferImpl.this.dropHead()
+      override def hasNext: Boolean = StashBufferImpl.this.nonEmpty
+      override def next(): T = StashBufferImpl.this.dropHead()
     }.take(numberOfMessages).map(wrap)
     val ctx = scaladslCtx.asInstanceOf[ActorContext[T]]
```

Comment:
this could just take in a Scala one for now but that would be inconsistent with other methods in Behavior